src/service/viewport/viewport-impl.js

```diff
@@ -875,7 +875,13 @@ export class ViewportImpl {
     this.lastPaddingTop_ = this.paddingTop_;
     this.paddingTop_ = paddingTop;
 
-    const animPromise = this.animateFixedElements_(duration, curve, transient);
+    const animPromise = this.fixedLayer_.animateFixedElements(
```

Comment:
is there a possibility of `updateOnViewportEvent_` being called when `fixedLayer_` is null?

Reply:
Based on the behavior of the previous code, I don't think there would be a possibility.
Previously, `animateFixedElements_` would call `this.fixedLayer_.updatePaddingTop` (and this call was not guarded by a null check), which will cause problems if `fixedLayer_` is null.

Reply:
Could the whole `updateOnViewportEvent_` be moved? My thought is the viewerHeader stuff is part of the viewer integration

Reply:
It is not possible at this time, due to `ViewportBindingDef`.